src/Language/PureScript/Parser/Types.hs

```diff
@@ -70,7 +70,7 @@ parseTypeAtom = indented *> P.choice
 
 parseConstrainedType :: TokenParser Type
 parseConstrainedType = do
-  constraints <- P.try (return <$> parseConstraint) <|> parens (commaSep1 parseConstraint)
+  constraints <- return <$> parseConstraint
```

Comment:
Yeah, I thought about that too. I'll clean this PR up in a little while and check I've not done anything silly.

One thing I'll relax is that currently parsing even a single constraint in parens will fail, but that seems unnecessary - you can wrap a type in parens, so why not constraints.